src/pages/waiting-room-to-car/cat-c/__tests__/waiting-room-to-car.cat-c.page.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { WaitingRoomToCarValidationError } from '../../waiting-room-to-car.actions';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { WarningBannerComponent } from '../../../../components/common/warning-banner/warning-banner';
+import { VehicleChecksCatCComponent } from '../components/vehicle-checks/vehicle-checks.cat-c';
+
+describe('WaitingRoomToCarCatCPage', () => {
+  let fixture: ComponentFixture<WaitingRoomToCarCatCPage>;
+  let component: WaitingRoomToCarCatCPage;
+  let store$: Store<StoreModel>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        WaitingRoomToCarCatCPage,
+        MockComponent(EyesightTestComponent),
+        MockComponent(EyesightFailureConfirmationComponent),
+        MockComponent(EndTestLinkComponent),
+        MockComponent(VehicleRegistrationComponent),
+        MockComponent(VehicleDetailsCardComponent),
+        MockComponent(VehicleDetailsComponent),
+        MockComponent(AccompanimentCardComponent),
+        MockComponent(AccompanimentComponent),
+        MockComponent(PracticeModeBanner),
+        MockComponent(VehicleChecksCatCComponent),
+        MockComponent(WarningBannerComponent),
+      ],
+      imports: [
+        IonicModule,
+        AppModule,
+        StoreModule.forRoot({
+          tests: () => ({
+            currentTest: {
+              slotId: '123',
+            },
+            testStatus: {},
+            startedTests: {
+              123: {
+                vehicleDetails: {},
+                accompaniment: {},
+                testData: {
+                  vehicleChecks: {
+                    tellMeQuestions: [],
+                    showMeQuestions: [],
+                  },
+                  seriousFaults: [],
+                  eyesightTest: {},
+                },
+                journalData: {
+                  candidate: {
+                    candidateName: 'Joe Bloggs',
+                  },
+                },
+              },
+            },
+          }),
+        }),
+      ],
+      providers: [
+        { provide: NavController, useFactory: () => NavControllerMock.instance() },
+        { provide: NavParams, useFactory: () => NavParamsMock.instance() },
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+        { provide: Platform, useFactory: () => PlatformMock.instance() },
+        { provide: AuthenticationProvider, useClass: AuthenticationProviderMock },
+        { provide: DateTimeProvider, useClass: DateTimeProviderMock },
+        { provide: QuestionProvider, useClass: QuestionProviderMock },
+      ],
+    })
+      .compileComponents()
+      .then(() => {
+        fixture = TestBed.createComponent(WaitingRoomToCarCatCPage);
+        component = fixture.componentInstance;
+      });
+    store$ = TestBed.get(Store);
+    spyOn(store$, 'dispatch');
+  }));
+
+  describe('DOM', () => {
+
+    describe('eyesight failure confirmation', () => {
```

Comment:
I guess we will be able to get rid of these eyesight tests later?

Reply:
Yeah, there is a story for implementing WRTC for cat c, which is going to take care of that